core/integration/secret_test.go

```diff
@@ -186,3 +189,105 @@ func (SecretSuite) TestBigScrubbed(ctx context.Context, t *testctx.T) {
 	require.NoError(t, err)
 	require.Equal(t, "***", stdout)
 }
+
+func (SecretSuite) TestVaultSecretsProviderTTL(ctx context.Context, t *testctx.T) {
```

Comment:
are there any chances of: 

- Make the test faster since it has to sleep at least 20s per case iteration
- Make the test more robust since it seems to be dependent on how much time things take to run?

Reply:
refactored the test, and found the sweet spot where it works.